pkg/api/controller.go

```diff
@@ -1423,7 +1423,7 @@ func (c *Controller) CreateRepository(w http.ResponseWriter, r *http.Request, bo
 	writeResponse(w, r, http.StatusCreated, response)
 }
 
-var errStorageNamespaceInUse = errors.New("lakeFS repositories can't share storage namespace")
+var errStorageNamespaceInUse = errors.New("lakeFS repositories can't share storage namespace. This includes re-using storage namespaces from repositories since deleted")
```

Comment:
Not sure we need to modify the error's text - we can keep it "storage namespace in use".
In the controller code I see that we already render a different text for the user - there you can format descriptive message.